src/components/ui/Highlight.svelte

```diff
@@ -1,11 +1,36 @@
 <svelte:options immutable={true} />
 
 <script lang="ts">
+  import { createEventDispatcher, onMount } from 'svelte';
+
   export let highlight: boolean = false;
+  export let flash: boolean = false;
```

Comment:
Is it necessary to dispatch a flash event if the HOC knows when `flash` is true?

Reply:
Here I'm dispatching the flash immediately to ensure that in case the user switches back to the span form before the animation completes that the animation and scroll to do not trigger again if the user eventually switches back to the directive form. `didFlash` makes sense to me though.